### PR TITLE
feat: allow defining signal type

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

this makes using especially the frontend API much usable as the signal can be typed instead having plain json object to play around with.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by drodil** [comment]
Not stale, waiting for review

File: plugins/signals-node/api-report.md [hunk 1]
```diff
@@ -11,19 +11,23 @@ import { ServiceRef } from '@backstage/backend-plugin-api';
...
   static create(options: SignalServiceOptions): DefaultSignalService;
-  publish(signal: SignalPayload): Promise<void>;
+  publish<SignalType extends JsonObject = JsonObject>(
+    signal: SignalPayload<SignalType>,
+  ): Promise<void>;
 }
 
 // @public (undocumented)
-export type SignalPayload = {
+export type SignalPayload<SignalType extends JsonObject = JsonObject> = {
```
Comment (freben) [inline plugins/signals-node/api-report.md:20]:
```suggestion
export type SignalPayload<TMessage extends JsonObject = JsonObject> = {
```

same throughout the PR

**Review by freben — COMMENTED** [review]
Sure, just a naming suggestion.

**Review by freben — APPROVED** [review]
👍

**Pull request merged**